src/controllers/article.controller.js

```diff
@@ -59,21 +59,36 @@ class ArticleController {
   static async getBookMarks(req, res) {
     const { id: userId } = req.currentUser;
 
-    const allBookMarks = await BaseRepository.findAndInclude(
-      db.Bookmark,
-      { userId },
-      db.Article,
-      'article'
-    );
+    const { page } = req.query;
+    const paginate = new Pagination(page, req.query.limit);
+    const { limit, offset } = paginate.getQueryMetadata();
 
-    if (allBookMarks.length < 1) {
-      return responseGenerator.sendError(
+    const {
+      count,
+      rows: allBookMarks
+    } = await BaseRepository.findCountAndInclude({
+      model: db.Bookmark,
+      options: { userId },
+      limit,
+      offset,
+      alias: 'article',
+      associatedModel: db.Article,
+      attributes: ['id', 'title', 'authorId', 'description']
+    });
+    if (count > 0) {
+      return responseGenerator.sendSuccess(
         res,
-        400,
-        `You currently do not have any article in your bookmark`
+        200,
+        null,
+        allBookMarks,
+        paginate.getPageMetadata(count, '/api/v1/articles')
       );
     }
-    return responseGenerator.sendSuccess(res, 200, null, allBookMarks);
+    return responseGenerator.sendError(
+      res,
+      400,
```

Comment:
The bookmarks list was **found** but it's an `empty` list, therefore it's not a `404`. The `Request` was well-formed, had no erroneous data, therefore it's not a `400 bad request`. 

This is a 200 and should return an empty array

Reply:
```
const message = count > 0 ? 'success messge' : 'no bookmarks'
return responseGenerator(res, statusCode, data, message, metadata)

```

Reply:
changes effected